Spotify-ObjC/Spotify-ObjC/SARequestManager.m

```diff
@@ -7,3 +7,87 @@
 //
 
 #import <Foundation/Foundation.h>
+#import "SARequestManager.h"
+
+@interface SARequestManager ()
+
+@end
+
+@implementation SARequestManager
+
+static NSURLSession *session;
+static SARequestManager *sharedInstance = nil;
+
+- (void) singletonInit {
+    NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
+    session = [NSURLSession sessionWithConfiguration: sessionConfig];
+}
+
++ (SARequestManager *) sharedInstance {
+    static dispatch_once_t dispatchOncePredicate = 0;
+    dispatch_once(&dispatchOncePredicate, ^{
+        sharedInstance = [[super allocWithZone:NULL] init];
+        [sharedInstance singletonInit];
+    });
+    return sharedInstance;
+}
+
++ (id) allocWithZone:(struct _NSZone *)zone {
+    return [SARequestManager sharedInstance];
+}
+
+- (void) getArtistsWithQuery:(NSString *)query completion:(Completion)completion {
+    NSMutableString *path = [NSMutableString stringWithFormat: @"https://api.spotify.com/v1/search?q=%@", query];
+    NSString *parameters = @"&type=track,artist&market=US";
+    [path appendString: parameters];
+    NSURL *url = [NSURL URLWithString: path];
+    NSURLSessionDataTask *task = [session dataTaskWithURL: url
+                                        completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                            SAResponse *result = [[SAResponse alloc] init];
+                                            if (!data) {
+                                                *(result.response) = Failure;
+                                                result.error = error;
+                                                NSLog(@"Error: %@", [error localizedDescription]);
+                                            } else {
+                                                *(result.response) = Success;
+                                                NSError *jsonError = nil;
+                                                NSDictionary *jsonResult = [NSJSONSerialization JSONObjectWithData: data
+                                                                                                           options: NSJSONReadingMutableContainers
+                                                                                                             error: &jsonError];
+                                                NSMutableArray *returnedArtists = [[NSMutableArray alloc] init];
+                                                
+                                                if ([jsonResult objectForKey:@"artists"]) {
+                                                    NSDictionary *artistResponse = [jsonResult objectForKey:@"artists"];
```

Comment:
SAArtist should have a `configureWithJSON` (or something) method where you can pass in the artist JSON object so that it can own the its own configuration rather than the RequestManager